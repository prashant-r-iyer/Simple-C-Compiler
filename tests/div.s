	.text

.globl main
main:
	# Save Frame pointer
	pushq %rbp
	movq %rsp, %rbp
	subq $256, %rsp
	pushq %rbx
	pushq %rbx
	pushq %r10
	pushq %r13
	pushq %r14
	pushq %r15
	pushq %r11
	pushq %r9
	movq $string0, %rbx

	# push 88
	movq $88, %r10

	# push 11
	movq $11, %r13

	# /
	movq %r10, %rax
	cqto
	idivq %r13
	movq %rax, %r10
	# func=printf nargs=2

	# Move values from reg stack to reg args
	movq %r10, %rsi
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx
	movq $string1, %rbx

	# push 83
	movq $83, %r10

	# push 11
	movq $11, %r13

	# %
	movq %r10, %rax
	cqto
	idivq %r13
	movq %rdx, %r10
	# func=printf nargs=2

	# Move values from reg stack to reg args
	movq %r10, %rsi
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

	# Restore registers
	popq %r9
	popq %r11
	popq %r15
	popq %r14
	popq %r13
	popq %r10
	popq %rbx
	popq %rbx
	addq $256, %rsp

	leave
	ret

string0:
	.string "88/11=%d\n"

string1:
	.string "83%11=%d\n"

