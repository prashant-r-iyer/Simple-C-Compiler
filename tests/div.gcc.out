div.c: In function ‘main’:
div.c:4:9: warning: implicit declaration of function ‘printf’ [-Wimplicit-function-declaration]
    4 |         printf("88/11=%d\n", 88/11);
      |         ^~~~~~
div.c:1:1: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
  +++ |+#include <stdio.h>
    1 | 
div.c:4:9: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
    4 |         printf("88/11=%d\n", 88/11);
      |         ^~~~~~
div.c:4:9: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
div.c:5:22: warning: unknown conversion type character ‘=’ in format [-Wformat=]
    5 |         printf("83%11=%d\n", 83%11);
      |                      ^
